src/main/java/com/telenordigital/sms/smpp/config/SmppConnectionConfig.java

```diff
@@ -85,6 +77,7 @@ public SmppConnectionConfig(String host, int port, int reconnectTimeSeconds) {
   }
 
   public URI connectionUrl() {
+    final var useTls = tls != null && tls.enabled();
```

Comment:
Maybe enabled flag is not needed since we are always checking for tls != null at the same time?